### PR TITLE
Bump IETF BLS spec version draft 03 -> draft 04

### PR DESCRIPTION
**Pull request by hwwhww**
Address #2072

### TODOs
- [x] Bump `py_ecc` to BLSv4 version
- [ ] Bump `milagro_bls_binding` to BLSv4 version

### CODE REVIEW

**Review by djrtwo — COMMENTED** [review]
This looks good to me

**Comment by djrtwo** [comment]
I modified this to point to `v1.0-candidate`. Separate from `dev` so that we can cleanly patch `v0.12` in the next few days if needbe

**Review by mratsim — APPROVED** [review]
LGTM as well

File: tests/generators/bls/main.py [hunk 1]
```diff
@@ -120,42 +145,46 @@ def case02_verify():
...
+    expect_exception(bls.Aggregate, [])
     # No signatures to aggregate. Follow IETF BLS spec, return `None` to represent INVALID.
-    # https://tools.ietf.org/html/draft-irtf-cfrg-bls-signature-02#section-2.8
+    # https://tools.ietf.org/html/draft-irtf-cfrg-bls-signature-04#section-2.8
     yield f'aggregate_na_signatures', {
         'input': [],
         'output': None,
     }
 
+    # Valid to aggregate G2 point at infinity
```
Comment (hwwhww) [inline tests/generators/bls/main.py:180]: I've opened an issue to get confirmation from IETF BLS standard maintainers: cfrg/draft-irtf-cfrg-bls-signature#31

**Review by melyalex — APPROVED** [review]

**Comment by hwwhww** [comment]
Two pending issues of milagro:
- https://github.com/sigp/milagro_bls/issues/39 for `Sign` with `SK==0` case.
- https://github.com/ChihChengLiang/milagro_bls_binding/issues/19 for `FastAggregateVerify` with the point at infinity `PK`.

I added some edge case handler in pyspec utils `tests/core/pyspec/eth2spec/utils/bls.py` for getting test generator ready ASAP. Once the pending issues are fixed, we can bump the next `milagro_bls_binding` version soon.

/cc @ChihChengLiang @kirk-baird @djrtwo
